skills/types/bool_file_test.go

```diff
@@ -0,0 +1,206 @@
+			"path": "/sys/class/leds/input27::capslock/brightness",
+		},
+	},
+	missingPathSkill: &skills.Skill{
+		Type: "bool-file",
+	},
+	badPathSkill: &skills.Skill{
+		Type:  "bool-file",
+		Attrs: map[string]interface{}{"path": "path"},
+	},
+	badTypeSkill: &skills.Skill{
+		Type: "other-type",
+	},
+	slot: &skills.Slot{
+		Type: "bool-file",
+	},
+	badTypeSlot: &skills.Slot{
+		Type: "other-type",
+	},
+})
+
+func (s *BoolFileTypeSuite) TestName(c *C) {
+	c.Assert(s.t.Name(), Equals, "bool-file")
+}
+
+func (s *BoolFileTypeSuite) TestSanitizeSkill(c *C) {
+	// Both LED and GPIO skills are accepted
+	err := s.t.SanitizeSkill(s.ledSkill)
+	c.Assert(err, IsNil)
+	err = s.t.SanitizeSkill(s.gpioSkill)
+	c.Assert(err, IsNil)
+	// Skills without the "path" attribute are rejected.
+	err = s.t.SanitizeSkill(s.missingPathSkill)
+	c.Assert(err, ErrorMatches,
+		"bool-file must contain the path attribute")
+	// Skills with incorrect value of the "path" attribute are rejected.
+	err = s.t.SanitizeSkill(s.badPathSkill)
+	c.Assert(err, ErrorMatches,
+		"bool-file can only point at LED brightness or GPIO value")
+	// It is impossible to use "bool-file" type to sanitize skills of other types.
+	c.Assert(func() { s.t.SanitizeSkill(s.badTypeSkill) }, PanicMatches,
+		`skill is not of type "bool-file"`)
+}
+
+func (s *BoolFileTypeSuite) TestSanitizeSlot(c *C) {
+	err := s.t.SanitizeSlot(s.slot)
+	c.Assert(err, IsNil)
+	// It is impossible to use "bool-file" type to sanitize slots of other types.
+	c.Assert(func() { s.t.SanitizeSlot(s.badTypeSlot) }, PanicMatches,
+		`skill slot is not of type "bool-file"`)
+}
+
+func (s *BoolFileTypeSuite) TestSlotSecuritySnippetHandlesSymlinkErrors(c *C) {
+	// Symbolic link traversal is handled correctly
+	types.MockEvalSymlinks(&s.BaseTest, func(path string) (string, error) {
+		return "", fmt.Errorf("broken symbolic link")
+	})
+	snippet, err := s.t.SlotSecuritySnippet(s.gpioSkill, skills.SecurityApparmor)
+	c.Assert(err, ErrorMatches, "cannot compute skill slot security snippet: broken symbolic link")
+	c.Assert(snippet, IsNil)
+}
+
+func (s *BoolFileTypeSuite) TestSlotSecuritySnippetDereferencesSymlinks(c *C) {
+	// Use a fake (successful) dereferencing function for the remainder of the test.
+	types.MockEvalSymlinks(&s.BaseTest, func(path string) (string, error) {
+		return "(dereferenced)" + path, nil
+	})
+	// Extra apparmor permission to access GPIO value
+	// The path uses dereferenced symbolic links.
+	snippet, err := s.t.SlotSecuritySnippet(s.gpioSkill, skills.SecurityApparmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, DeepEquals, []byte(
+		"(dereferenced)/sys/class/gpio/gpio13/value rwk,\n"))
+	// Extra apparmor permission to access LED brightness.
+	// The path uses dereferenced symbolic links.
+	snippet, err = s.t.SlotSecuritySnippet(s.ledSkill, skills.SecurityApparmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, DeepEquals, []byte(
+		"(dereferenced)/sys/class/leds/input27::capslock/brightness rwk,\n"))
```

Comment:
Done, can you check if that is what you had on your mind? Look at ``TestSlotSecurityDoesNotContainSkillSecurity``